packages/video_player/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayerPlugin.java

```diff
@@ -84,144 +104,95 @@ public void onCancel(Object o) {
-            @Override
-            public boolean onError(MediaPlayer mp, int what, int extra) {
-              if (eventSink != null) {
-                eventSink.error(
-                    "VideoError", "Video player had error " + what + " extra " + extra, null);
-              }
-              return true;
-            }
-          });
+      exoPlayer.addListener(
+          new DefaultEventListener() {
 
-      mediaPlayer.setOnCompletionListener(
-          new MediaPlayer.OnCompletionListener() {
             @Override
-            public void onCompletion(MediaPlayer mediaPlayer) {
-              Map<String, Object> event = new HashMap<>();
-              event.put("event", "completed");
-              eventSink.success(event);
+            public void onPlayerStateChanged(final boolean playWhenReady, final int playbackState) {
+              super.onPlayerStateChanged(playWhenReady, playbackState);
+              if (playbackState == Player.STATE_BUFFERING) {
+                if (eventSink != null) {
+                  Map<String, Object> event = new HashMap<>();
+                  event.put("event", "bufferingUpdate");
+                  List<Integer> range = Arrays.asList(0, exoPlayer.getBufferedPercentage());
+                  // iOS supports a list of buffered ranges, so here is a list with a single range.
+                  event.put("values", Collections.singletonList(range));
+                  eventSink.success(event);
+                }
+              } else if (playbackState == Player.STATE_READY && !isInitialized) {
+                isInitialized = true;
+                sendInitialized();
+              }
             }
-          });
 
-      mediaPlayer.setOnInfoListener(
-          new MediaPlayer.OnInfoListener() {
             @Override
-            public boolean onInfo(MediaPlayer mediaPlayer, int what, int extra) {
-              Map<String, Object> event = new HashMap<>();
-              switch (what) {
-                case MediaPlayer.MEDIA_INFO_BUFFERING_START:
-                  {
-                    event.put("event", "bufferingStart");
-                    eventSink.success(event);
-                    return true;
-                  }
-                case MediaPlayer.MEDIA_INFO_BUFFERING_END:
-                  {
-                    event.put("event", "bufferingEnd");
-                    eventSink.success(event);
-                    return true;
-                  }
+            public void onPlayerError(final ExoPlaybackException error) {
+              super.onPlayerError(error);
+              if (eventSink != null) {
+                eventSink.error("VideoError", "Video player had error " + error, null);
               }
-              return false;
             }
           });
 
-      mediaPlayer.prepareAsync();
-
       Map<String, Object> reply = new HashMap<>();
       reply.put("textureId", textureEntry.id());
       result.success(reply);
     }
 
-    @SuppressWarnings("deprecation")
```

Comment:
Is there no ExoPlayer equivalent of the code below?

Reply:
There actually is. I thought this was just a workaround to ensure proper audio playback on older Android devices. I readded this.